### PR TITLE
Add limit for the maximum frequency of a task

### LINKED ISSUE #179: Frequency parser throws unhandled exception for large numbers
Command:
image

Log:
image

UI does not update to indicate that the frequency is too large.
Expected Result: a `CommandException` to be thrown indicating the malformed input.

### PR DESCRIPTION
**Pull request by ranaldmiao**
Limits frequency to between 0 and 999 inclusive.

Can be changed by modifying FREQUENCY_LIMIT in Frequency.java.

Fixes #179

### CODE REVIEW

File: src/main/java/seedu/address/model/task/Frequency.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import static java.util.Objects.requireNonNull;
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
+import java.math.BigInteger;
```
Comment (kc97ble) [inline src/main/java/seedu/address/model/task/Frequency.java]: Why?? hahaha!

File: src/main/java/seedu/address/model/task/Frequency.java [hunk 2]
```diff
@@ -14,9 +15,9 @@
  */
 public class Frequency implements Comparable<Frequency> {
 
-
+    public static final Integer FREQUENCY_LIMIT = 1000; //Strictly less than
     public static final String MESSAGE_FREQUENCY_CONSTRAINTS =
-        "Frequency should only be a non negative number";
+        "Frequency should only be a non negative integer not exceeding " + FREQUENCY_LIMIT.toString();
```
Comment (kc97ble) [inline src/main/java/seedu/address/model/task/Frequency.java]: Contradiction:
Line 18: Frequency must be strictly smaller than 1000
Line 20: Frequency must be no more than 1000

File: src/main/java/seedu/address/model/task/Frequency.java [hunk 3]
```diff
@@ -14,9 +15,9 @@
...
     public static final String FREQUENCY_VALIDATION_REGEX = "[0-9]+";
```
Comment (kc97ble) [inline src/main/java/seedu/address/model/task/Frequency.java]: Use this one: "\d{1,3}"
It matches from 1 to 3 digits.

File: src/test/java/seedu/address/model/task/FrequencyTest.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
+package seedu.address.model.task;
```
Comment (kc97ble) [inline src/test/java/seedu/address/model/task/FrequencyTest.java]: wow, awesome work!

File: docs/UserGuide.adoc [hunk 5]
```diff
@@ -95,7 +95,7 @@ dates are: 29/03/2018, 29/3/2018, 1/1/2018.
 * `PRIORITY` should only be 0, 1, 2, 3, or 4.
 The lower the number, the higher the priority of the task.
 (1 = Highest Priority, 4 = Lowest Priority)
-* `FREQUENCY` should be a non-negative integer (number of days).
+* `FREQUENCY` should be a non-negative integer (number of days) not exceeding 1000.
```
Comment (kc97ble) [inline docs/UserGuide.adoc]: Can you revert the change (to avoid conflict)? I updated UG already.
In my UG:

> FREQUENCY should be a non-negative integer between 0 and 999 (number of days)

Comment (ranaldmiao) [reply docs/UserGuide.adoc]: Sure. Will do so

File: src/main/java/seedu/address/model/task/Frequency.java (hunk 1 above)
Comment (ranaldmiao) [reply src/main/java/seedu/address/model/task/Frequency.java]: To parse arbitrary large integers that the less restrictive regex allows.

File: src/main/java/seedu/address/model/task/Frequency.java (hunk 2 above)
Comment (ranaldmiao) [reply src/main/java/seedu/address/model/task/Frequency.java]: Right! Thanks for spotting this. Will change line 20.

File: src/main/java/seedu/address/model/task/Frequency.java (hunk 3 above)
Comment (ranaldmiao) [reply src/main/java/seedu/address/model/task/Frequency.java]: Hmmm, will this match and work for f/00005 ?

Comment (ranaldmiao) [reply src/main/java/seedu/address/model/task/Frequency.java]: I think I'll keep to using [0-9]+ and use BigInteger to restrict the value.

File: docs/UserGuide.adoc (hunk 5 above)
Comment (ranaldmiao) [reply docs/UserGuide.adoc]: I have merged master and #320 into this PR.
Do help look through and see if its ok.

**Comment by ranaldmiao** [comment]
@kc97ble I have updated, do look through again and provide comments.
